ts/features/bonus/cgn/components/merchants/discount/CgnBucketCodeComponent.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react";
+import { View } from "native-base";
+import { TouchableWithoutFeedback } from "@gorhom/bottom-sheet";
+import { StyleSheet } from "react-native";
+import { Millisecond } from "italia-ts-commons/lib/units";
+import I18n from "../../../../../../i18n";
+import { BaseTypography } from "../../../../../../components/core/typography/BaseTypography";
+import { IOColors } from "../../../../../../components/core/variables/IOColors";
+import { useIODispatch, useIOSelector } from "../../../../../../store/hooks";
+import { isLoading, isReady } from "../../../../bpd/model/RemoteValue";
+import Eye from "../../../../../../../img/icons/Eye.svg";
+import ActivityIndicator from "../../../../../../components/ui/ActivityIndicator";
+import { cgnBucketSelector } from "../../../store/reducers/bucket";
+import { cgnCodeFromBucket } from "../../../store/actions/bucket";
+import { addEvery } from "../../../../../../utils/strings";
+import IconFont from "../../../../../../components/ui/IconFont";
+import { clipboardSetStringWithFeedback } from "../../../../../../utils/clipboard";
+
+const styles = StyleSheet.create({
+  row: {
+    flexDirection: "row"
+  },
+  codeContainer: { alignItems: "center", justifyContent: "space-between" },
+  codeText: {
+    fontSize: 20
+  },
+  flexEnd: { alignSelf: "flex-end" }
+});
+
+const COPY_ICON_SIZE = 24;
+const FEEDBACK_TIMEOUT = 3000 as Millisecond;
+
+const CgnBucketCodeComponent = () => {
+  const [isCodeVisible, setIsCodeVisible] = React.useState(false);
+  const [isTap, setIsTap] = React.useState(false);
+  const timerRetry = React.useRef<number | undefined>(undefined);
+
+  const dispatch = useIODispatch();
+  const code = useIOSelector(cgnBucketSelector);
+
+  const requestBucket = () => {
+    dispatch(cgnCodeFromBucket.request());
+    setIsCodeVisible(true);
+  };
+
+  const handleCopyPress = () => {
+    if (isReady(code)) {
+      setIsTap(true);
+      clipboardSetStringWithFeedback(code.value.code);
+      // eslint-disable-next-line functional/immutable-data
+      timerRetry.current = setTimeout(() => setIsTap(false), FEEDBACK_TIMEOUT);
```

Comment:
this PR LGTM
the only thing I would add is an effect that returns the callback to clean the timeout. It could be useful when the user dismisses the view before the timer is triggered (<3 sec)

```typescript
useEffect(
    () => () => {
      if (timerRetry.current) {
        clearTimeout(timerRetry.current);
      }
    },
    []
  );
```
what do you think?

Reply:
Sure, just forgot to add before. 
Included in 880f56f